codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerCombinedErrorGenerator.kt

```diff
@@ -30,12 +35,35 @@ open class ServerCombinedErrorGenerator(
     private val symbolProvider: RustSymbolProvider,
     private val operation: OperationShape
 ) {
-    private val operationIndex = OperationIndex.of(model)
-
-    open fun render(writer: RustWriter) {
-        val errors = operationIndex.getErrors(operation)
-        val operationSymbol = symbolProvider.toSymbol(operation)
+    fun render(writer: RustWriter) {
+        val errors = operation.operationErrors(model)
         val symbol = operation.errorSymbol(symbolProvider)
+        val operationSymbol = symbolProvider.toSymbol(operation)
+        if (errors.isNotEmpty()) {
+            renderErrors(writer, errors.map { it.asStructureShape().get() }, symbol, operationSymbol)
+        }
+
+        if (operation.isEventStream(model)) {
+            operation.eventStreamErrors(model)
+                .forEach { (unionShape, unionErrors) ->
+                    if (unionErrors.isNotEmpty()) {
+                        renderErrors(
+                            writer,
+                            unionErrors,
+                            unionShape.eventStreamErrorSymbol(symbolProvider),
+                            symbolProvider.toSymbol(unionShape)
+                        )
+                    }
+                }
+        }
+    }
+
+    open fun renderErrors(
+        writer: RustWriter,
+        errors: List<StructureShape>,
+        errorSymbol: RuntimeType,
+        operationSymbol: Symbol
+    ) {
```

Comment:
It doesn't look like, but please let's be sure the changes on this function will not break the Python implementation. The `render()` function signature has changed, so that will have to be fixed.

Reply:
`render()` hasn't changed actually, if you were using `render()` it should keep working the same way

Reply:
Ok, great! We are using directly `render()`